projects/testing-library/src/lib/models.ts

```diff
@@ -210,6 +232,24 @@ export interface RenderComponentOptions<ComponentType, Q extends Queries = typeo
    * })
    */
   componentInputs?: Partial<ComponentType> | { [alias: string]: unknown };
+
+  /**
+   * @description
+   * An object to set `@Input` or `input()` properties of the component
+   *
+   * @default
+   * {}
+   *
+   * @example
+   * await render(AppComponent, {
+   * inputs: {
+   *  counterValue: 10,
+   *  // explicitly define aliases this way:
+   *  someAlias: aliasedInputWithValue('value')
+   * })
+   */
+  inputs?: ComponentInput<ComponentType>;
```

Comment:
Hi, thanks for your input!
After giving it some thought I think we don't need a branded type.
We can utilize typescript to infer the correct type if the property exists, or else accept any type for an unknown property.

```suggestion
  inputs?: {[K in keyof ComponentType]?: ComponentType[K] extends Signal<infer U> ? U : ComponentType[K]} & {[K in Exclude<string, keyof ComponentType>]: unknown};
```

Reply:
I think that aliased types are very uncommon in practice, and there is some merit in needing to explicitly mark them as such.

With this approach, if you refactor an input, you won't receive any errors until you run the test. Or users can misspell them and not realize.

Is this something you considered?

Reply:
I believe the behavior will be the same, or am I missing something?

- normal inputs: need to have the same type as defined in the component
- aliased inputs: can be whatever. There's no check if the property name exists

E.g. `myName` needs to be a string and throws an error if it's not

![image](https://github.com/user-attachments/assets/79601178-b0ba-49e1-b322-c946089666ed)

Reply:
It's your call at the end of the day, but allowing unchecked aliases will lead to leftovers of inputs that no longer exist, just to allow for a feature that the vast majority of the user base doesn't need.

I would personally much rather prefer strict type checking to allow in not passing any extraneous inputs, or that if I refactor an input and rename it, I instantly see all the places that need to be updated, at compile time.

The second case in your example is an explicit user override, think of it like the `$any()` type cast function. Once you use it, all guarantees are off.

We could rename some things still in this PR. That's up for bikeshedding.

Reply:
> It's your call at the end of the day, but allowing unchecked aliases will lead to leftovers of inputs that no longer exist, just to allow for a feature that the vast majority of the user base doesn't need.

I agree with this. 💯

> I would personally much rather prefer strict type checking that ensures no extraneous inputs. And that if I refactor an input and rename it, I instantly see all the places that need to be updated, at compile time.

I only don't see how `aliasInputWithValue` helps to prevent this.

As far as I see we get the same benefits from the proposed solutions:


| | `aliasInputWithValue` | TS Types |
|--|--|--|
| Type-safety for normal inputs (throws when given value is not the correct type) | ✅ | ✅ |
| Inputs are not required (e.g. it's possible to not set an input) | ✅ | ✅ |
| Flexibility for aliased inputs | ✅ | ✅ |
| Property renames in the component are propagated to `inputs` | ✅ | ✅ |
| The possibility to add non-existing properties | ✅ | ✅ | 

In terms of DX (imho), I think the TS types are easier to use.
The behavior for inputs with a transform seems also to be the same.

Reply:
Edit: nvm, I was wrong. My IDE was acting weird. I'm revisiting this.